planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/scene.cpp

```diff
@@ -1392,130 +1396,90 @@ bool NormalLaneChange::check_candidate_path_safety(
+  const auto max_lane_changing_length = std::invoke([&]() {
+    double max_length = transient_data.dist_to_terminal_end - prep_metric.length;
+    max_length =
+      std::min(max_length, dist_lc_start_to_end_of_lanes - transient_data.next_dist_buffer.min);
+    return max_length;
+  });
 
-  if (!is_valid_start_point(common_data_ptr_, lane_changing_start_pose.value())) {
-    RCLCPP_DEBUG(
-      logger_,
-      "Reject: lane changing points are not inside of the target preferred lanes or its "
-      "neighbors");
-    return {};
-  }
+  const auto max_path_velocity = prepare_segment.points.back().point.longitudinal_velocity_mps;
+  constexpr double lane_changing_lon_accel{0.0};
+  const auto lane_changing_metrics = calculation::calc_shift_phase_metrics(
+    common_data_ptr_, shift_length, prep_metric.velocity, max_path_velocity,
+    lane_changing_lon_accel, max_lane_changing_length);
 
-  const auto resample_interval = utils::lane_change::calcLaneChangeResampleInterval(
-    current_min_dist_buffer, minimum_lane_changing_velocity);
-  const auto target_lane_reference_path = utils::lane_change::get_reference_path_from_target_Lane(
-    common_data_ptr_, lane_changing_start_pose.value(), current_min_dist_buffer, resample_interval);
+  const auto sorted_lane_ids = utils::lane_change::get_sorted_lane_ids(common_data_ptr_);
 
-  if (target_lane_reference_path.points.empty()) {
-    RCLCPP_DEBUG(logger_, "Reject: target_lane_reference_path is empty!!");
-    return {};
+  LaneChangePath candidate_path;
+  for (const auto & lc_metric : lane_changing_metrics) {
+    try {
+      candidate_path = get_candidate_path(
+        prep_metric, lc_metric, prepare_segment, sorted_lane_ids, lane_changing_start_pose,
+        shift_length);
+    } catch (const std::exception & e) {
+      continue;
+    }
+    terminal_lane_change_path_ = candidate_path;
+    return candidate_path.path;
   }
 
-  lane_change_info.shift_line = utils::lane_change::get_lane_changing_shift_line(
-    lane_changing_start_pose.value(), target_segment.points.front().point.pose,
-    target_lane_reference_path, shift_length);
-
-  auto reference_segment = prev_module_output_.path;
-  const double length_to_lane_changing_start = autoware::motion_utils::calcSignedArcLength(
-    reference_segment.points, reference_segment.points.front().point.pose.position,
-    lane_changing_start_pose->position);
-  utils::clipPathLength(reference_segment, 0, length_to_lane_changing_start, 0.0);
-  // remove terminal points because utils::clipPathLength() calculates extra long path
-  reference_segment.points.pop_back();
-  reference_segment.points.back().point.longitudinal_velocity_mps =
-    static_cast<float>(minimum_lane_changing_velocity);
-
-  const auto terminal_lane_change_path = utils::lane_change::construct_candidate_path(
-    common_data_ptr_, lane_change_info, reference_segment, target_lane_reference_path,
-    sorted_lane_ids);
-
-  return terminal_lane_change_path;
+  return {};
```

Comment:
to be explicit, return `std::nullopt` instead
```suggestion
  return std::nullopt;
```

Reply:
fixed in [fb2f9e0](https://github.com/autowarefoundation/autoware.universe/pull/9592/commits/fb2f9e0178d8befbbd3710f12f13e745a2605b2c)